MBBSEmu.Tests/CPU/FADD_Tests.cs

```diff
@@ -0,0 +1,36 @@
+﻿using Iced.Intel;
+using System;
+using Xunit;
+using static Iced.Intel.AssemblerRegisters;
+
+namespace MBBSEmu.Tests.CPU
+{
+    public class FADD_Tests : CpuTestBase
+    {
+        [Theory]
+        [InlineData(0.5, 1.5)]
+        [InlineData(float.MaxValue, 1)]
+        [InlineData(float.MinValue, float.MaxValue)]
+        [InlineData(0, 0)]
+        public void FADD_Test(float initialST0Value, float valueToAdd)
+        {
+            Reset();
+
+            CreateDataSegment(new ReadOnlySpan<byte>(), 2);
```

Comment:
does this return the segment? (I thought I made that change).

I'd find it easier to understand if you stored the segment return value and used it in the calls to SetArray and DS = <val>

Reply:
It doesn't -- it just creates an empty segment at the specified ordinal. You have the option to pass in the data to load the segment with initially, which in this case is just nothing.